parsers/NED.py

```diff
@@ -241,7 +241,11 @@ def fetch_production(
     json_data = call_api(target_datetime)
     NED_data = format_data(json_data, logger)
 
-    all_dates = [item.get("datetime") for item in NED_data.to_list()]
+    all_dates = [
+        item.get("datetime")
+        for item in NED_data.to_list()
+        if item.get("datetime") is not None
```

Comment:
To me it should be but VSCode still complains. But moving the check down to the next statement seems to do the trick if that works for you.

I think it's due to the multiple get calls and the type checker simply is not smart enough to get that. By moving it down it checks just a single variable (instead of getting twice). Should have the same end result right?

```python
    all_dates = [item.get("datetime") for item in NED_data.to_list()]

    if all(
        date >= datetime(2021, 1, 1, tzinfo=timezone.utc)
        for date in all_dates
        if date is not None # Add it here instead.
    ):
        return NED_data.to_list()
```

Reply:
Yes, that should do the trick!